run/run.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/python3
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import configparser
+import gzip
+import json
+import platform as host_platform
+import re
+import requests
+import subprocess
+import sys
+import os
+
+"""
+run.py runs WPT and uploads results to Google Cloud Storage.
+
+The dependencies setup and running portion of this script should intentionally
+be left small. The brunt of the work should take place in WPT's `wptrun`:
+https://github.com/w3c/web-platform-tests/blob/master/tools/wptrun.py
+
+# Running the script
+
+Before you run the script, you need to:
+
+1. Copy run/running.example.ini to run/running.ini
+2. Modify the applicable fields of run/running.ini (this may also involve installing browsers)
+3. Make sure you have the correct secret in run/running.ini
+4. Install dependencies with `pip3 install -r requirements.txt`
+5. Make sure you have gsutil installed (see https://cloud.google.com/storage/docs/gsutil)
```

Comment:
So when this is run on Sauce, or Chromium infrastructure, or somewhere else, I suppose these steps (including getting the secret token) are all required?